client/components/swipeable/index.js

```diff
@@ -0,0 +1,290 @@
+import classnames from 'classnames';
+import { useRtl } from 'i18n-calypso';
+import { Children, useState, useEffect, useRef, useCallback } from 'react';
+
+import './style.scss';
+
+const OFFSET_THRESHOLD_PERCENTAGE = 0.45; // Percentage of width to travel before we trigger the slider to move to the desired slide.
+const VELOCITY_THRESHOLD = 0.5; // Speed of drag above, before we trigger the slider to move to the desired slide.
+
+function useUpdateLayout( enabled, currentPageIndex, updateLayout ) {
+	// save callback to a ref so that it doesn't need to be a dependency of other hooks
+	const savedUpdateLayout = useRef();
+	useEffect( () => {
+		savedUpdateLayout.current = updateLayout;
+	}, [ updateLayout ] );
+
+	// fire when the `currentPageIndex` parameter changes
+	useEffect( () => {
+		if ( ! enabled ) {
+			return;
+		}
+
+		savedUpdateLayout.current();
+	}, [ enabled, currentPageIndex ] );
+
+	// fire when the window resizes
+	useEffect( () => {
+		if ( ! enabled ) {
+			return;
+		}
+
+		const onResize = () => savedUpdateLayout.current( false );
+		window.addEventListener( 'resize', onResize );
+		window.addEventListener( 'orientationchange', onResize );
+		return () => {
+			window.removeEventListener( 'resize', onResize );
+			window.removeEventListener( 'orientationchange', onResize );
+		};
+	}, [ enabled ] );
+}
+
+function getDragPositionAndTime( event ) {
+	const { timeStamp } = event;
+	if ( event.hasOwnProperty( 'clientX' ) ) {
+		return { x: event.clientX, y: event.clientY, timeStamp };
+	}
+
+	if ( event.targetTouches[ 0 ] ) {
+		return {
+			x: event.targetTouches[ 0 ].clientX,
+			y: event.targetTouches[ 0 ].clientY,
+			timeStamp,
+		};
+	}
+
+	const touch = event.changedTouches[ 0 ];
+	return { x: touch.clientX, y: touch.clientY, timeStamp };
+}
+
+function getPagesWidth( pageWidth, numPages ) {
+	if ( ! pageWidth ) {
+		return null;
+	}
+	return pageWidth * numPages;
+}
+
+export const Swipeable = ( {
+	hasDynamicHeight = false,
+	children,
+	currentPage = 0,
+	onPageSelect,
+	pageClassName,
+	containerClassName,
+	...otherProps
+} ) => {
+	const [ pageWidth, setPageWidth ] = useState( null );
+	const [ swipeableArea, setSwipeableArea ] = useState();
+	const isRtl = useRtl();
+
+	const [ pagesStyle, setPagesStyle ] = useState( {
+		transform: `translate3d(0px, 0px, 0px)`,
+		transitionDuration: `300ms`,
+	} );
+	const [ dragStartData, setDragStartData ] = useState( null );
+
+	const pagesRef = useRef();
+	const resizeObserverRef = useRef();
+	const numPages = Children.count( children );
+
+	const getOffset = useCallback(
+		( index ) => {
+			const offset = pageWidth * index;
```

Comment:
`pageWidth` starts as `null`, and could be technically also set to `undefined` when calling `setPageWidth( resizeObserverRef.current?.getBoundingClientRect().width );`

Therefore, it would be best to check for it not to be `null` or `undefined` before using it a as a number value. The same check should be performed also before using `pageWidth` in `handleDragEnd`.